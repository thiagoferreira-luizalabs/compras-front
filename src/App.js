import React, { Component } from 'react';
import './App.css';
import 'semantic-ui-css/semantic.min.css';
import { Menu, Button } from 'semantic-ui-react'


class App extends Component {
  render() {
    return (
      <div className="App">
        <Menu>
          <Menu.Item header>Compras</Menu.Item>
          <Menu.Item name='aboutUs' />
          <Menu.Item name='jobs' />
          <Menu.Item name='locations' />
          <Menu.Item>
            <Button primary>Sign up</Button>
          </Menu.Item>

          <Menu.Item>
            <Button>Log-in</Button>
          </Menu.Item>
        </Menu>
      </div>
    );
  }
}

export default App;
