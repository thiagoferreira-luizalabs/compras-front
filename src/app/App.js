import React, { Component } from 'react';
import PageHeader, { ItemsMenu } from "./components/page-header/PageHeader"
import { Menus } from "./routes/Pages"
import { Menu, Segment, Sidebar } from 'semantic-ui-react'
import { Link } from "react-router-dom"

class App extends Component {

  state = { visible: false }

  handleButtonClick = () => this.setState({ visible: !this.state.visible })

  handleSidebarHide = () => this.setState({ visible: false })

  render() {
    const { isAuthenticated } = this.props.auth
    const { visible } = this.state

    return (
      <div className="app">
        {isAuthenticated && <PageHeader menus={Menus} onToggleSideBar={this.handleButtonClick} />}

        <Sidebar.Pushable as={Segment}>
          <Sidebar as={Menu} animation='overlay' icon='labeled' inverted onHide={this.handleSidebarHide} vertical visible={visible} width='thin' >
            {ItemsMenu(Menus)}
            <Menu.Item as={Link} to="/logout">Logout</Menu.Item>
          </Sidebar>

          <Sidebar.Pusher>
            <Segment basic>
              {this.props.children}
            </Segment>
          </Sidebar.Pusher>
        </Sidebar.Pushable>


      </div>
    );
  }
}

export default App;
