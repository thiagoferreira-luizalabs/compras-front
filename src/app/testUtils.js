import React from "react"
import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

const MemoryRouterHOC = (children) => (<MemoryRouter>{children}</MemoryRouter>)

const toJsonSnapshot = (wrapper) => toJson(wrapper, { mode: "deep", noKey: true })

export {
  shallow, mount, sinon, MemoryRouter, toJsonSnapshot, asyncAssert, createFileChangeEvent, MemoryRouterHOC
}
