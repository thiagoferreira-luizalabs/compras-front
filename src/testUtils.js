import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
// import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

export {
  shallow, mount, sinon,  /*MemoryRouter,*/ toJson, asyncAssert, createFileChangeEvent
}
