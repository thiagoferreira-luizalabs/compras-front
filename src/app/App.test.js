import React from 'react';
import App from './App';
import { mount, toJsonSnapshot, MemoryRouterHOC } from "./testUtils"

it("Renders public page without menu", () => {
  const props = {
    auth: {
      isAuthenticated: false,
      user: null,
      token: null
    }
  }
  let wrapper = mount(MemoryRouterHOC(<App {...props}>Testing</App>))
  expect(toJsonSnapshot(wrapper)).toMatchSnapshot()
})

it("Renders private page with menu", () => {
  const props = {
    auth: {
      isAuthenticated: true,
      user: "Tester",
      token: "12321312"
    }
  }
  let wrapper = mount(MemoryRouterHOC(<App {...props}>Testing</App>))
  expect(toJsonSnapshot(wrapper)).toMatchSnapshot()
})
