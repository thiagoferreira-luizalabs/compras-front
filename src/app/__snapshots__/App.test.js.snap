// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders private page with menu 1`] = `
<div
  className="app"
>
  <div>
    <div>
      <div
        className="strips"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        className="clear"
      />
    </div>
    <div
      className="ui topbar-mobile menu"
    >
      <a
        className="item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="bars icon"
        />
      </a>
      <a
        className="item logo"
        href="/home"
        onClick={[Function]}
      >
        <img
          className="ui image"
          src="magazineluiza.png"
        />
      </a>
    </div>
    <div
      className="ui navbar menu"
    >
      <a
        className="item"
        href="/"
        onClick={[Function]}
      >
        <img
          className="ui image logo"
          src="magazineluiza.png"
        />
      </a>
      <div
        aria-expanded={false}
        className="ui item dropdown active"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Comercial
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Logística
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Tema
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/theme"
            onClick={[Function]}
            role="option"
          >
            Tema
          </a>
        </div>
      </div>
      <div
        className="right menu logout"
      >
        <div
          aria-expanded={false}
          className="ui item dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Olá, CBT_SILVA
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
          >
            <a
              className="item"
              href="/logout"
              onClick={[Function]}
              role="option"
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui segment pushable"
  >
    <div
      className="ui inverted vertical labeled icon ui overlay left thin sidebar menu"
    >
      <div
        aria-expanded={false}
        className="ui item dropdown active"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Comercial
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Logística
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Tema
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/theme"
            onClick={[Function]}
            role="option"
          >
            Tema
          </a>
        </div>
      </div>
      <a
        className="item"
        href="/logout"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
    <div
      className="pusher"
    >
      <div
        className="ui basic segment"
      >
        Testing
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders public page without menu 1`] = `
<div
  className="app"
>
  <div
    className="ui segment pushable"
  >
    <div
      className="ui inverted vertical labeled icon ui overlay left thin sidebar menu"
    >
      <div
        aria-expanded={false}
        className="ui item dropdown active"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Comercial
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Logística
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
          <a
            className="item"
            href="/home"
            onClick={[Function]}
            role="option"
          >
            Inicio
          </a>
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Tema
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/theme"
            onClick={[Function]}
            role="option"
          >
            Tema
          </a>
        </div>
      </div>
      <a
        className="item"
        href="/logout"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
    <div
      className="pusher"
    >
      <div
        className="ui basic segment"
      >
        Testing
      </div>
    </div>
  </div>
</div>
`;
